unstructured/metrics/object_detection.py

```diff
@@ -0,0 +1,665 @@
+        Initializes the ObjectDetection prediction and ground truth,
+        and converts the data to the required format.
+
+        Args:
+            prediction_file_path (Path): path to json file with predictions dump from OD model
+            ground_truth_file_path (Path): path to json file with OD ground truth data
+        """
+        # TODO: Test after https://unstructured-ai.atlassian.net/browse/ML-92 is done.
+        with open(prediction_file_path) as f:
+            predictions_data = json.load(f)
+        with open(ground_truth_file_path) as f:
+            ground_truth_data = json.load(f)
+
+        assert (
+            predictions_data["object_detection_classes"]
+            == ground_truth_data["object_detection_classes"]
+        ), "Classes in predictions and ground truth do not match."
+        assert len(predictions_data["pages"]) == len(
+            ground_truth_data["pages"]
+        ), "Pages number in predictions and ground truth do not match."
+        for pred_page, gt_page in zip(predictions_data["pages"], ground_truth_data["pages"]):
+            assert (
+                pred_page["size"] == gt_page["size"]
+            ), "Page sizes in predictions and ground truth do not match."
+
+        class_labels = predictions_data["object_detection_classes"]
+        document_preds = cls._process_data(predictions_data, class_labels, prediction=True)
+        document_targets = cls._process_data(ground_truth_data, class_labels)
+        pages_height, pages_width = cls._parse_page_dimensions(predictions_data)
+
+        return cls(document_preds, document_targets, pages_height, pages_width, class_labels)
+
+    @staticmethod
+    def _parse_page_dimensions(data: dict) -> tuple[list, list]:
+        """
+        Process the page dimensions from the json file to the required format.
+        """
+        pages_height = []
+        pages_width = []
+        for page in data["pages"]:
+            pages_height.append(page["size"]["height"])
+            pages_width.append(page["size"]["width"])
+        return pages_height, pages_width
+
+    @staticmethod
+    def _process_data(data: dict, class_labels, prediction: bool = False) -> list[dict]:
+        """
+        Process the elements from the json file to the required format.
+        """
+        pages_list = []
+        for page in data["pages"]:
+            page_elements = []
+            for element in page["elements"]:
+                # Extract coordinates, confidence, and class label from each prediction
+                class_label = element["type"]
+                class_idx = class_labels.index(class_label)
+                if prediction:
+                    confidence = element["prob"]
+                    x1, y1, x2, y2 = element["bbox"]
+                    page_elements.append([x1, y1, x2, y2, confidence, class_idx])
+                else:
+                    x, y, w, h = element["bbox"]
+                    page_elements.append([class_idx, x, y, w, h])
```

Comment:
is the order here correct? Above the coordinates are first?

Reply:
yes, in super-gradients their functions use two different formats for gt and pred